snuba/query/processors/mandatory_condition_applier.py

```diff
@@ -0,0 +1,22 @@
+from snuba.query.logical import Query
+from snuba.clickhouse.processors import QueryProcessor
+from snuba.request.request_settings import RequestSettings
+
+from snuba.request import Request
+from snuba.datasets.dataset import Dataset
+from snuba.query.conditions import combine_and_conditions
+
+
+class MandatoryConditionApplier(QueryProcessor):
+    def process_query(self, query: Query, request_settings: RequestSettings) -> None:
+
+        query_plan = Dataset.get_query_plan_builder().build_plan(Request)
```

Comment:
You do not need to ask a query object to the dataset. You are already receiving it as a parameter of this method (see line 11).

More details about this.
A query processor is not supposed to depend on datasets. That would create a circular dependency since datasets already depend on query processors.
That's why query processor are provided a query and request_settings. That is everything they should depend on to per form the processing work.